pagopa/io-app
# fix: [IABT-1368] Wrong message router navigation from push notification

## Short description
When the app is opened from a push notification, it uses the non-paginated version of the MessageRouterScreen. This loads all the messages from the non-paginated APIs, causing an inconsistent state in the store.

#### I think this navigation should be refactored in order to avoid spreading this responsibility throughout the app. The purpose of this PR is to fix the bug, but a more clean approach will be used once we remove the legacy code.

## List of changes proposed in this pull request
- Check for pagination before calling `navigateToMessageRouterScreen`

## How to test
Send a push notification to the device or the simulator. Open the app from the notification and after reading the message go back to the messages list. You should see all you messages. Please check it with both the app killed and suspended in background.

https://user-images.githubusercontent.com/467098/162386545-7c682296-f1b0-4a46-93db-767baf267af2.MP4

https://user-images.githubusercontent.com/467098/162386587-77f526c6-b279-4c1f-960e-8ec67a8eb3fc.mp4
